docs/developer-guide/debugging.md

```diff
@@ -117,3 +117,31 @@ code .           // Starts VS Code in the current directory
 
 After VS Code has started, you can set breakpoints in the Goblint code. You can start debugging using the VS Code command palette with the command `"Debug: Start Debugging"`, or alternatively by pressing `F5`. Note that the Goblint execution is considerably slowed down in the debugger, so you have to be somewhat patient.
 Of course, don't forget to rebuild the debuggable executable if you make changes to the code!
+
+## Debugging Issues with Larger Programs
+
+Sometimes during development one may encounter instances where, e.g., the verifying phase reports that the fixpoint is not reached. This is usually due to bugs in `join`, `widen` or `leq`. For small programs, one can find the cause by inspecting the program and the output carefully. If the issue happens only with large programs it is hard to understand.
+
+To work on such cases, it makes sense to reduce the program to a small example program that still triggers the same issue. This can either be done by hand or using `creduce`. `creduce` takes as input a script that terminates with status `0` if the reduced program is still interesting and the program.
```

Comment:
>if the reduced program is still interesting and the program.

There seems to be something missing?

Reply:
no, the sentence is just a bit awkward. It gets as arguments the script and the original program.

Reply:
Ah, I see.